### DISCUSSION TITLE
Enable more state changes via code

### DISCUSSION THREAD

**Discussion by dgounaris** [gilbarbara/react-spotify-web-playback#113] [Ideas]
Currently, if I understand the documentation correctly, only the play/pause can be controlled through code.

I think it could be useful to have some react state exposed that would allow for updates to more playback status aspects (i.e. go to next track, change volume), to increase extensibility of this library and allow custom automations to handle the playback device.

Just to illustrate a very simple use case to try and prove my point, someone would then be able to change the volume of this playback device programmatically based on whether the tab is the one active or not, which is not directly possible now through this library.

**Comment by KobiAms** [comment]
or maybe receiving a ref and forwarding it to the Spotify player itself to control it manually if the ui not needed (in my use-case)

**Reply by gilbarbara** [reply]
Hey @KobiAms 
If you need the UI, you don't need this package since 90% of the bundle size is the components.
Just set up the [web-playback-sdk](https://developer.spotify.com/documentation/web-playback-sdk) directly in your app.

**Reply by KobiAms** [reply]
Hey @gilbarbara.
First of all thank you for your quick response.
I must say that I really like this package because it make the connection with Spotify-Web-Playback very straight forward and stable.
I Try to use the SDK but your package make it a lot more easier as a react wrapper.
thanks a lot anyway you helped me a lot.

**Comment by gilbarbara** [comment]
Hey,

With the latest update, version 0.14, you now have access to the `getPlayer` prop which enables direct control of the player. Additionally, the new `components` prop lets you include extra buttons to the player.
Furthermore, the Spotify API methods are now exported, giving you the ability to manage device functionality like repeat and shuffle.
